### DISCUSSION TITLE
`EnsureChannelFirstd` without Metadata

### DISCUSSION THREAD

**Discussion by bhashemian** [Project-MONAI/MONAI#4931] [Q&A]
It is very common that the raw data does not include any metadata, and transforming it to metadata will not include necessary metadata for following transforms. For instance, consider the case below that the input is numpy, and `EnsureChannelFirstd` depends on the metadata to be able to correctly apply the transform. How this can be achieved without using any deprecated components (`AsChannelFirstd`) or deprecated arguments?

I did not find any utility components to add metadata after using `ToMetaTensord`. Is there any way to achieve this?

```python
from monai.data import DataLoader, Dataset
from monai.transforms import Compose, EnsureChannelFirstd, ToMetaTensord
from monai.utils import first
train_preprocess = Compose(
    [
        ToMetaTensord(keys=("image", )),
        EnsureChannelFirstd(keys=("image", )),
    ]
)

train_data = [{"image": image} for image in np.random.rand(10, 5, 5, 3)]
print(train_data[0]['image'].shape)

train_dataset = Dataset(
    data=train_data,
    transform=train_preprocess,
)

train_dataloader = DataLoader(train_dataset, batch_size=4)
print(first(train_dataloader)["image"].shape)
```

```
(5, 5, 3)
(4, 1, 5, 5, 3)
```

**Comment by bhashemian** [comment]
@rijobro @wyli @Nic-Ma I would appreciate your help here.

**Reply by wyli** [reply]
thanks, I think this is an issue and I'd suggest an API update: https://github.com/Project-MONAI/MONAI/pull/4933

**Reply by wyli** [reply]
for this specific case a workaround would be manually creating the metadict `train_data = [{"image": image, "image_meta_dict": {"original_channel_dim": 0}} for image in np.random.rand(10, 5, 5, 3)]`

**Reply by bhashemian** [reply]
Thanks, that makes sense.
